mono/metadata/marshal-ilgen.c

```diff
@@ -1485,7 +1485,16 @@ emit_invoke_call (MonoMethodBuilder *mb, MonoMethod *method,
 	}
 
 	if (sig->ret->byref) {
+		int pos;
+
 		/* perform indirect load and return by value */
+#ifdef ENABLE_NETCORE
+		mono_mb_emit_byte (mb, CEE_DUP);
+		pos = mono_mb_emit_branch (mb, CEE_BRTRUE);
+		mono_mb_emit_exception_full (mb, "Mono", "NullByRefReturnException", NULL);
```

Comment:
Will this `throw` break the user into the debugger if they have first chance exceptions turned on?

In .NET Native we had to make this return a special sentinel value that we check for and then do the throw with the correct exception because multiple first chance exceptions tend to be confusing to users.

Reply:
> Will this throw break the user into the debugger if they have first chance exceptions turned on?

I think it will. If not here then definitely at the site in https://github.com/mono/mono/pull/14128/files#diff-04cafe842355fb36dd025dfc83aa13be that converts it to regular `NullReferenceException`.

> In .NET Native we had to make this return a special sentinel value that we check for and then do the throw with the correct exception because multiple first chance exceptions tend to be confusing to users.

Ha, I had exactly the same idea how to implement this.